### PR TITLE
open .pom file in RO mode

### LINKED ISSUE #718: open POM under local maven repo in read-only mode
It's a follow-up item of #428 

So far vscode has no such API, but we can workaround it via ContentProvider API, the same way we show effectivePOM and dependency list now.

### PR DESCRIPTION
**Pull request by Eskibear**
fix #718 

image

### CODE REVIEW

File: src/contentProvider.ts [hunk 1]
```diff
@@ -8,6 +8,12 @@ import { MavenProject } from "./explorer/model/MavenProject";
 import { getDependencyTree } from "./handlers/showDependenciesHandler";
 import { Utils } from "./utils/Utils";
 
+/**
+ * URI patterns.
+ * vscode-maven://dependencies/<pom-path>/Dependencies?<pom-path>
+ * vscode-maven://effective-pom/<pom-path>/EffectivePOM.xml?<pom-path>
+ * vscode-maven:///<pom-path-in-local-maven-repository>
```
Comment (jdneo) [inline src/contentProvider.ts:15]: three `/`s here, as expected?

Comment (CsCherrYY) [inline src/contentProvider.ts:15]: how about using a specific `authority` like the previous ones?

Comment (Eskibear) [reply src/contentProvider.ts:15]: as expected. Here authority is empty, so `//<empty>` is authority. And `path` starts with "/".

> URI = scheme:[//authority]path[?query][#fragment]

I use `authority` to differentiate dependencies/effective-pom/common pom file.

Comment (Eskibear) [reply src/contentProvider.ts:15]: yes, it's an option.

Comment (Eskibear) [reply src/contentProvider.ts:15]: Let's call it "local-repository"? What if later we want to open a project pom.xml in RO mode?

**Review by CsCherrYY — APPROVED** [review]

File: src/contentProvider.ts (hunk 1 above)
Comment (Eskibear) [reply src/contentProvider.ts:15]: > What if later we want to open a project pom.xml in RO mode?

no such requirement for the moment. let's rename / use another authority if later we need  do that.

**Pull request merged**